django/forms/widgets.py

```diff
@@ -573,6 +573,24 @@ def render_options(self, selected_choices):
                 output.append(self.render_option(selected_choices, option_value, option_label))
         return '\n'.join(output)
 
+    def choice_has_empty_value(self, choice):
+        choice_value = choice[0]
+        if isinstance(choice_value, six.string_types):
+            return len(choice_value) == 0
+        return choice_value is None
+
+    def use_required_attribute(self, initial):
+        # If first item has a value, rendering required attribute causes HTML
+        # validation error.
+        super_result = super(Select, self).use_required_attribute(initial)
+
+        # Do checks only for select itself, not for its children
```

Comment:
This is a bit unclear to me. When you say "for select itself" do you mean the `Select` widget class? Does "its children" mean subclasses? `NullBooleanSelect` seems to be at least one subclass that doesn't pass these checks.